uptycs/data_source_tag.go

```diff
@@ -0,0 +1,227 @@
+package uptycs
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/uptycslabs/uptycs-client-go/uptycs"
+)
+
+type dataSourceTagType struct {
+	p Provider
+}
+
+func (r dataSourceTagType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"value": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"key": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"flag_profile_id": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"custom_profile_id": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"compliance_profile_id": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"process_block_rule_id": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"dns_block_rule_id": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"windows_defender_preference_id": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"tag": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"custom": {
+				Type:     types.BoolType,
+				Optional: true,
```

Comment:
Can we add the planmodifier here as well, like a few of the other files have?

Reply:
I don't know what the default value is so i havent added any planmodifiers we havent run into

Reply:
The default values in all of these are undocumented so i havent added any planmodifiers we havent run into

Reply:
these are data sources so you cant provide it. it will have a value. what youre reading is a bit misleading. It cant not have a value.